src/main/java/io/stargate/sgv2/jsonapi/service/shredding/collections/WritableShreddedDocument.java

```diff
@@ -304,6 +297,28 @@ public void shredVector(JsonPath path, ArrayNode vector) {
       queryVectorValues = arrayVals;
     }
 
+    @Override
+    public void shredVector(JsonPath path, ObjectNode vector) {
+      // vector data is added only to queryVectorValues and exists keys index
+      addKey(path);
+
+      String binaryString = vector.get(JsonExtensionType.BINARY.encodedName()).asText();
+
+      byte[] binaryPayload;
+      binaryPayload = Base64Variants.MIME_NO_LINEFEEDS.decode(binaryString);
```

Comment:
That makes sense, but we'll probably want to catch exception from decoder in case invalid String is passed (and there needs to be test(s) for it).